### PR TITLE
Add indeterminate state for checkbox

### PR DESCRIPTION
**Pull request by oorestisime**
Signed-off-by: Orestis Ioannou <oorestisime@gmail.com>

#### What does this PR do?

Add indeterminate state to the CheckBox component

#### Where should the reviewer start?

I would say the storybook for the rendering part. Maybe the color of the line should be neutral?

#### What testing has been done on this PR?

* it renders correctly
* warning when used with toggle checkboxes
* warning when both checked and indeterminate state is activated

#### What are the relevant issues?

#1323

#### Do the grommet docs need to be updated?

Covered in the PR

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible

### CODE REVIEW

File: src/js/components/CheckBox/CheckBox.js [hunk 1]
```diff
@@ -39,7 +40,22 @@ class CheckBox extends Component {
       hidden = <input name={name} type="hidden" value="true" />;
     }
 
-    const Icon = theme.checkBox.icons.checked;
+    if (checked && indeterminate) {
```
Comment (alansouzati) [inline src/js/components/CheckBox/CheckBox.js]: can we do this in the constructor instead? I don't think it is a good idea to check this in every render function (called multiple times).

**Comment by alansouzati** [comment]
screen shot 2018-11-27 at 10 18 43 pm

can we avoid the line to take all the width ?

I think this will align better with the other icons:

`<path fill="none" d="M6,12 L18,12"></path>`

File: src/js/components/CheckBox/CheckBox.js (hunk 1 above)
Comment (oorestisime) [reply src/js/components/CheckBox/CheckBox.js]: yes thanks didn't think of that!

**Comment by oorestisime** [comment]
Done and rebased with master

**Pull request merged**